File: scala-project/src/main/scala/com/waitingforcode/Writer.scala
package com.waitingforcode

import org.apache.spark.sql.{DataFrame, Row}
import org.apache.spark.sql.streaming.{DataStreamWriter, OutputMode, Trigger}

object Writer {

  def setUpSessionsWriter(sessionsToWrite: DataFrame,
                          trigger: Trigger): DataStreamWriter[Row] = {
    sessionsToWrite.writeStream.outputMode(OutputMode.Append)
      .trigger(trigger)
  }

}

File: python-project/visits_generator/writer.py
from typing import Dict

from pyspark.sql import DataFrame
from pyspark.sql.streaming import DataStreamWriter


def set_up_sessions_writer(sessions_to_write: DataFrame, trigger: Dict[str, str]) -> DataStreamWriter:
    return (sessions_to_write.writeStream
            .outputMode("append")
            .trigger(**trigger))
